### PR TITLE
Resolves #91 add scratchy service

### LINKED ISSUE #91: [Front] Create scratchy service that contains methods for backend requests access.
Methods for REST API in scratchy service (javascript) for:

- [x] Message

- [x] User

- [x] Room

Methods are:

- [x] get by id

- [x] update by id

- [x] delete by id

- [x] get all

Using axios

### PR DESCRIPTION
**Pull request by yannis-mlgrn**
Fix #91

### CODE REVIEW

**Review by Benvii — CHANGES_REQUESTED** [review]
Thank for your contribution, as your branch isn't up to date, can you rebase it ?

You can improve it using JSDoc typeDef for more correct type documentation with more details, some of the return types indicated are incorrect.

File: front/js/service.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+//                               ROOM FUNCTIONS
+    /**
+     * @param {string} roomID - room id , eg:60895dd62d1a706830c31f10
+     * @param {string} roomTitle - title of a room , eg:my room
+     * @param {string} roomDescription - description of a room, eg:my room description
+     * @returns {string} - room id
+     */
+    this.updateRoom = async function(roomId,roomTitle,roomDescription){
+        const reponse = await axios.put(apiUrl+'/room/'+roomId, { title: roomTitle , description: roomDescription });
```
Comment (Draeggon) [inline front/js/service.js:28]: shoudn't the id be in the response ?

File: front/js/service.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        return reponse.data;
+    };
+
+    /**
+     * 
+     * @param {string} roomID - room id , eg: 60895dd62d1a706830c31f10
+     */
+    this.deleteRoom = async function(roomID){
+        const reponse = await axios.delete(apiUrl+'/room/'+roomID);
+        return reponse.data;
```
Comment (Draeggon) [inline front/js/service.js:49]: the response is an empty json do we really need to return it ?

File: front/js/service.js (hunk 1 above)
Comment (yannis-mlgrn) [reply front/js/service.js:28]: yes, the id is in the request. I said that in line 25 : ```  @returns {string} - room id  ```


screen

File: front/js/service.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
+/*
+- add axios cdn link in index.html
+     	<scritp src="https://unpkg.com/axios@0.21.1/dist/axios.min.js"></scritp>
```
[the remark this answers is not in the document]
Comment (Benvii) [reply front/js/service.js:3]: Still a typo but no matter it's not a big problem

**Review by Benvii — CHANGES_REQUESTED** [review]
Changes are still need i re-opened the conversations

**Pull request closed without merging**